packages/integration-karma/test/shadow-dom/event-in-shadow-tree/event-propagation.spec.js

```diff
@@ -213,27 +225,39 @@ describe('non-composed and bubbling event propagation in nested shadow tree', ()
                 new CustomEvent('test', { composed: false, bubbles: true })
             );
 
-            const composedPath = [
-                nodes['span-manual'],
-                nodes['div-manual'],
-                nodes['x-shadow-tree'].shadowRoot,
-            ];
-            const expectedLogs = [
-                [nodes['span-manual'], nodes['span-manual'], composedPath],
-                [nodes['div-manual'], nodes['span-manual'], composedPath],
-                [nodes['x-shadow-tree'].shadowRoot, nodes['span-manual'], composedPath],
-            ];
-            if (!process.env.NATIVE_SHADOW) {
-                // TODO [#1569]: Listeners on the following targets should not be invoked when the event is non-composed.
-                expectedLogs.push(
-                    [document.body, null, composedPath],
-                    [document.documentElement, null, composedPath],
-                    [document, null, composedPath]
-                );
-            }
             expect(logs).toEqual(expectedLogs);
         });
     });
+
+    describe('when non-composed event crosses immediate shadow root boundary', () => {
+        beforeEach(() => {
+            setFeatureFlagForTest('ENABLE_NON_COMPOSED_EVENTS_LEAKAGE', true);
+        });
+        afterEach(() => {
+            setFeatureFlagForTest('ENABLE_NON_COMPOSED_EVENTS_LEAKAGE', false);
+        });
+
+        it('propagate event from a child element added via lwc:dom="manual"', () => {
+            // Fire the event in next macrotask to allow time for the MO to key the manually inserted nodes
+            return new Promise((resolve) => {
+                setTimeout(resolve);
+            }).then(() => {
+                const logs = dispatchEventWithLog(
+                    nodes['span-manual'],
+                    new CustomEvent('test', { composed: false, bubbles: true })
+                );
+
+                if (!process.env.NATIVE_SHADOW) {
+                    expectedLogs.push(
+                        [document.body, null, composedPath],
+                        [document.documentElement, null, composedPath],
+                        [document, null, composedPath]
+                    );
```

Comment:
This test is a little difficult to follow. I think we should wrap the entire `describe()` with this `if (!process.env.NATIVE_SHADOW)` and use the complete set of expected logs:
```suggestion
                    const expectedLogs = [
                        [nodes['span-manual'], nodes['span-manual'], composedPath],
                        [nodes['div-manual'], nodes['span-manual'], composedPath],
                        [nodes['x-shadow-tree'].shadowRoot, nodes['span-manual'], composedPath],
                        [document.body, null, composedPath],
                        [document.documentElement, null, composedPath],
                        [document, null, composedPath],
                    ];
```